src/common/blkdev.cc

```diff
@@ -545,10 +545,20 @@ std::string get_device_id(const std::string& devname,
   if (blkdev.serial(buf, sizeof(buf))) {
     serial = buf;
   }
-  if (!model.size() || serial.size()) {
+  if (!model.size()) {
```

Comment:
i am not sure if this is the intended behavior. i think what we really want is to bail out of the case where `model.empty() && !serial.empty()`.

**if** we want to return an empty device id if either of them is empty, could put something like

```c++
if (err) {
  if (model.empty() && serial.empty()) {
    *err = ...;
    return {};
  } else if (model.empty()) {
    *err = ...;
    return {};
  } else if (serial.empty()) {
    *err = ...;
    return {};
  }
}
```

less indent this way.

Reply:
Yeah i think the original intent was just for ``||`` to be ``&&``.  That is probably the only case that I saw, but unfortunately I think this fallback logic was written prior to switching to using udev for everything, so it doesn't cover as many cases.

Reply:
Thanks, I've fixed logic here.

Another discovery is on line 545 -- `if (blkdev.serial(buf, sizeof(buf))) {` looks wrong.